ports/esp32/mphalport.c

```diff
@@ -112,6 +112,12 @@ uintptr_t mp_hal_stdio_poll(uintptr_t poll_flags) {
 int mp_hal_stdin_rx_chr(void) {
     for (;;) {
         int c = ringbuf_get(&stdin_ringbuf);
+        #if CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG
+        if (c == -1) {
+            usb_serial_jtag_handle_rx();
```

Comment:
Only checking for pending Serial/JTAG data when the ringbuffer is empty will limit the throughput, and may still leave the risk of a race condition losing data if `mp_hal_stdin_rx_char()` isn't called enough before another serial data packet is received (i.e. if the ring buffer is mostly empty but not entirely empty when the packet arrives).

Suggest taking a look at the `poll_cdc_interfaces()` function on ports with a USB-CDC port (i.e. rp2, samd). I think it'd be possible to implement this as a `usb_serial_jtag_poll_rx()` and call it from the same places that `poll_cdc_interfaces()` is called in other ports. From a maintainers perspective, this also has the advantage of keeping the pattern similar. What do you think?